pants-plugins/internal_plugins/versioned_setup/register.py

```diff
@@ -0,0 +1,81 @@
+import os.path
+
+from pants.backend.python.goals.setup_py import SetupKwargs, SetupKwargsRequest
+from pants.engine.fs import DigestContents, GlobMatchErrorBehavior, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+
+
+class VersionedSetupKwargsRequest(SetupKwargsRequest):
+    @classmethod
+    def is_applicable(cls, _: Target) -> bool:
+        # We always use our custom `setup()` kwargs generator for `python_distribution`
+        # targets in this repo.
+        return True
+
+
+@rule
+async def setup_kwargs_plugin(request: VersionedSetupKwargsRequest) -> SetupKwargs:
+    kwargs = request.explicit_kwargs.copy()
+
+    # Add classifiers. We preserve any that were already set.
+    standard_classifiers = [
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: MIT",
+        "Operating System :: MacOS :: MacOS X",
+        "Operating System :: POSIX :: Linux",
+        "Programming Language :: Python",
+        "Topic :: Software Development :: Build Tools",
+    ]
+    kwargs["classifiers"] = [*standard_classifiers, *kwargs.get("classifiers", [])]
+    project_urls = {
+        "Source": "https://github.com/CodeForAfrica/api",
+        "Tracker": "https://github.com/CodeForAfrica/api/issues",
+        "Twitter": "https://twitter.com/Code4Africa",
+    }
+    kwargs["project_urls"] = {**project_urls, **kwargs.get("project_urls", {})}
+    default_kwargs = dict(
+        url="https://github.com/CodeForAfrica/api",
+        author="Code for Africa",
+        author_email="tech@codeforafrica.org",
+        license="MIT",
+        zip_safe=True,
+    )
+    for key, value in default_kwargs.items():
+        if key not in kwargs:
+            kwargs[key] = value
+
+    # version can be set to directly or via version_file relative to the BUILD file.
+    version = kwargs.get("version", None)
+    # version_file is not a standard kwarg, hence we need to pop it from kwargs.
+    version_file = kwargs.pop("version_file", None)
+    if version and version_file:
+        raise ValueError(
+            f"The python_distribution target {request.target.address} has supplied both "
+            "`version` and `version_file` in its setup_py's kwargs. Only one of these "
+            "should be supplied."
+        )
```

Comment:
@kilemensi what if version kwargs and version_file content are similar? Should we still raise this error

Reply:
The question then becomes can you think of a _legitimately_ good reason why one would want to to set both of these to the same value at the same time @KhadijaMahanga?

Reply:
@kilemensi No legitimate reason 🚀